### DISCUSSION TITLE
Lenovo Lenovo K14 Gen 1 on ArchLinux not firmware upgrade

### DISCUSSION THREAD

**Discussion by kenmatrix** [fwupd/fwupd#6546] [General]
I'm using archlinux and so far I haven't been able to update the firmware. It always appears that it will update and nothing happens after that.
This is the first time I've opened a discussion here

**Comment by hughsie** [comment]
> I'm using archlinux and so far I haven't been able to update the firmware

What happens (in detail) when you use `fwupdmgr update`?

**Comment by kenmatrix** [comment]
[cassio@archlinux ~]$ fwupdmgr update
Dispositivos com nenhuma atualização de firmware disponível:
 • ELAN0678:00 04F3:31B8
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
 • UEFI Device Firmware
Dispositivos com a versão mais recente do firmware disponível:
 • SSSTC CL4-4D256-Q79
╔══════════════════════════════════════════════════════════════════════════════╗
║ Atualizar System Firmware de 0.1.13 para 0.1.27?                             ║
╠══════════════════════════════════════════════════════════════════════════════╣
║ This stable release fixes the following issues:                              ║
║                                                                              ║
║ • Fix cannot access lenovo cloud server.                                     ║
║ • Fixed an issue where running WmiOpcodePasswordAdmin twice would return     ║
║ an error                                                                     ║
║                                                                              ║
║ 21CUS0TN00 deve permanecer conectado a uma fonte de energia durante a        ║
║ atualização para evitar danos.                                               ║
╚══════════════════════════════════════════════════════════════════════════════╝
Efetuar a operação? [Y|n]: 
Aguardando…              [***************************************] Menos que um minuto restante…
Firmware instalado com sucesso
Do not turn off your computer or remove the AC adapter while the update is in progress.
 • UEFI dbx
Uma atualização requer uma reinicialização para ser concluída. Reiniciar agora? [y|N]:

**Reply by kenmatrix** [reply]
Upon restarting, the update process begins, you are informed that the update will begin and the screen flashes and the system starts normally

**Reply by kenmatrix** [reply]
If you prefer, I can video you can do the upgrade process

**Reply by hughsie** [reply]
Does it work if UEFI secure boot is turned off?

**Reply by kenmatrix** [reply]
No

**Reply by kenmatrix** [reply]
I disabled Secure Boot and tried to do the update and everything happened the same way I already inform you

**Reply by hughsie** [reply]
> you are informed that the update will begin

Can you take a picture of this text please? It sounds like the update is being scheduled fine, but not starting properly.

Also, we don't really test using archlinux, you might be better asking in an archlinux-specific forum. Does it work with Fedora or Ubuntu perhaps?

**Reply by kenmatrix** [reply]
https://youtu.be/fCDVPFaf1DI

I made a video showing the update

**Reply by hughsie** [reply]
Okay, the update is being scheduled properly, it's just erroring out after starting. @mrhpearson any ideas?

**Reply by mrhpearson** [reply]
I'm guessing that this is a GUID clash - the K14 isn't a certified laptop (to my knowledge) and I don't think we have any FW on LVFS for it. (and the latest version on the Lenovo site is 1.16)

Can you post the output from 'fwupdmgr get-devices' so I can check and see what you are picking up please?

Probably best not to apply this update for now!

**Reply by kenmatrix** [reply]
[cassio@archlinux ~]$ fwupdmgr get-devices 
LENOVO 21CUS0TN00
│
├─ELAN0678:00 04F3:31B8:
│     ID do dispositivo:  f8bebdac7d9634f3890f6f6cf3eec0908c2c700b
│     Resumo:             Touchpad
│     Versão atual:       0x0001
│     Versão do gerenciador de boot:0x0002
│     Fornecedor:         ELAN Microelectronics (HIDRAW:0x04F3)
│     GUIDs:              0692c70e-5fe8-56b1-a4b1-ff406903ce06 ← HIDRAW\VEN_04F3&DEV_31B8
│                         417fc220-c06b-56d6-a11e-2a04d658bab8 ← HIDRAW\VEN_04F3&DEV_31B8&MOD_010F
│                         400648c0-5ab8-5b67-964a-b9c1ce664e5e ← ELANTP\ICTYPE_14&MOD_010F
│                         077ac9fb-3a80-520a-8d63-9dbb50eabdd8 ← ELANTP\ICTYPE_14&MOD_010F&DRIVER_HID
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│     Device Requests:    • Message
│                         • Image
│   
├─AMD Ryzen 3 5400U with Radeon Graphics:
│ │   ID do dispositivo:  4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│ │   Versão atual:       0x0a50000d
│ │   Fornecedor:         Advanced Micro Devices, Inc.
│ │   GUIDs:              79759cdc-94db-5098-be7b-eb02521fbbec ← CPUID\PRO_0&FAM_19&MOD_50
│ │                       20b595b0-5892-5870-8e4c-688133ad6e34 ← CPUID\PRO_0&FAM_19&MOD_50&STP_0
│ │   Opções do dispositivo:• Dispositivo interno
│ │   Device Requests:    • Message
│ │ 
│ └─Secure Processor:
│       ID do dispositivo:c54ab0237d7a8db8c717b68e0be78e4374a2a079
│       Versão atual:     00.11.00.81
│       Versão do gerenciador de boot:00.11.00.81
│       Fornecedor:       Advanced Micro Devices, Inc. (PCI:0x1022)
│       GUIDs:            0e8dc554-a0a2-51fb-b439-1eb72b14ec38 ← PCI\VEN_1022&DEV_15DF
│                         1b2da1bd-0375-538c-855c-b2cf4479cefc ← PCI\VEN_1022&DEV_15DF&SUBSYS_17AA50AB
│       Opções do dispositivo:• Dispositivo interno
│       Device Requests:  • Message
│     
├─SSSTC CL4-4D256-Q79:
│     ID do dispositivo:  3743975ad7f64f8d6575a9ae49fb3a8856fe186f
│     Resumo:             NVM Express solid state drive
│     Versão atual:       4U879
│     Fornecedor:         Solid State Storage Technology Corporation (NVME:0x1E95)
│     Número de série:    SS1F28691L4BR38S51WA
│     GUIDs:              f0ef5984-a8fd-5d25-a34d-39b359aa1c4c ← NVME\VEN_1E95&DEV_1007
│                         0989d431-7fee-5b86-a1e0-a9bcc1e4da2c ← NVME\VEN_1E95&DEV_1007&SUBSYS_126F2267
│                         f79bc3aa-d1ca-5e0f-9587-0beedb63f92b ← SSSTC CL4-4D256-Q79
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • O sistema exige uma fonte de alimentação externa
│                         • Suporte no servidor remoto
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│                         • Carga útil assinada
│     Device Requests:    • Message
│                         • Image
│                         • Image (custom)
│   
├─System Firmware:
│ │   ID do dispositivo:  a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Resumo:             UEFI ESRT device
│ │   Versão atual:       0.1.13
│ │   Fornecedor:         Lenovo (DMI:LENOVO)
│ │   Estado da atualização:Falhou
│ │   Erro de atualização:failed to update to 0: error-auth-error
│ │   GUID:               1d68c247-db50-4759-b641-27ac698c1f5a
│ │   Opções do dispositivo:• Dispositivo interno
│ │                       • Atualizável
│ │                       • O sistema exige uma fonte de alimentação externa
│ │                       • Suporte no servidor remoto
│ │                       • Precisa de uma reinicialização após a instalação
│ │                       • A verificação criptográfica de hash está disponível
│ │                       • O dispositivo pode ser usado durante a atualização
│ │   Device Requests:    • Message
│ │                       • Image
│ │                       • Image (custom)
│ │ 
│ └─UEFI dbx:
│       ID do dispositivo:362301da643102b9f38477387e2193e57abaa590
│       Resumo:           UEFI revocation database
│       Versão atual:     371
│       Versão mínima:    371
│       Fornecedor:       UEFI:Linux Foundation
│       Duração de instalação:1 segundo
│       GUIDs:            efd5c776-30c5-566c-9128-8f5f0012f2c7 ← UEFI\CRT_BF509BEDEB2135C96B9CB45A77B4A806E01CFF106F14F7CD99F9CB323213DA16&ARCH_X64
│                         f8ba2887-9411-5c36-9cee-88995bb39731 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503&ARCH_X64
│       Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • Suporte no servidor remoto
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│                         • Apenas atualizações de versão são permitidas
│                         • Carga útil assinada
│       Device Requests:  • Message
│                         • Image
│     
├─TPM:
│     ID do dispositivo:  c6a80ac3a22083423992a3cb15018989f37834d6
│     Versão atual:       7.2.3.0
│     Fornecedor:         Nuvoton Technology (TPM:NTC)
│     GUIDs:              fac1c8f3-73c8-5cd6-8330-07a3690b5140 ← TPM\VEN_NTC&DEV_0000
│                         2b17590c-d755-5c13-a7d4-f68fb0dcb4c4 ← TPM\VEN_NTC&MOD_NPCT75x"!"4rls
│                         e9ccc1dc-960a-5e09-afe9-e59a904b776d ← TPM\VEN_NTC&DEV_0000&VER_2.0
│                         5c410147-6fa3-5a6c-9386-c006959c7f91 ← TPM\VEN_NTC&MOD_NPCT75x"!"4rls&VER_2.0
│     Opções do dispositivo:• Dispositivo interno
│                         • O sistema exige uma fonte de alimentação externa
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode recuperar falhas de flashing
│                         • Os segredos de criptografia de disco completa podem ser invalidados ao atualizar
│                         • Carga útil assinada
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     ID do dispositivo:  349bb341230b1a86e5effe7dfe4337e1590227bd
│     Resumo:             UEFI ESRT device
│     Versão atual:       2
│     Fornecedor:         DMI:LENOVO
│     Estado da atualização:Sucesso
│     GUID:               152d8237-891c-4c65-a51b-605432e65688
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • O sistema exige uma fonte de alimentação externa
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│     Device Requests:    • Message
│                         • Image
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     ID do dispositivo:  2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Resumo:             UEFI ESRT device
│     Versão atual:       16842757
│     Versão mínima:      1
│     Fornecedor:         DMI:LENOVO
│     Estado da atualização:Sucesso
│     GUID:               c0342eb2-15b2-4fbd-9263-f6dcd69befae
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • O sistema exige uma fonte de alimentação externa
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│     Device Requests:    • Message
│                         • Image
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     ID do dispositivo:  f95c9218acd12697af946874bfe4239587209232
│     Resumo:             UEFI ESRT device
│     Versão atual:       65542
│     Versão mínima:      65542
│     Fornecedor:         DMI:LENOVO
│     Estado da atualização:Sucesso
│     GUID:               4bea12df-56e3-4cdb-97dd-f133768c9051
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • O sistema exige uma fonte de alimentação externa
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│     Device Requests:    • Message
│                         • Image
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     ID do dispositivo:  d96de5c124b60ed6241ebcb6bb2c839cb5580786
│     Resumo:             UEFI ESRT device
│     Versão atual:       117572352
│     Versão mínima:      117572352
│     Fornecedor:         DMI:LENOVO
│     Estado da atualização:Sucesso
│     GUID:               3954e118-d997-4499-b917-d4c454e4b124
│     Opções do dispositivo:• Dispositivo interno
│                         • Atualizável
│                         • O sistema exige uma fonte de alimentação externa
│                         • Precisa de uma reinicialização após a instalação
│                         • O dispositivo pode ser usado durante a atualização
│     Device Requests:    • Message
│                         • Image
│                         • Image (custom)
│
